### ISSUE TITLE
Another silly question this time about Items Add New Item

### ISSUE THREAD

**Issue by odiea** [jekkos/opensourcepos#551]
I placed this into item form. It works on update but does not auto fill when adding to a new form. Any suggestion? It would be nice not to have the help put in the suggested retail price.
<?php echo form_input(array(
							'name'=>'unit_price',
							'id'=>'unit_price',
							'class'=>'form-control input-sm',
							'value'=>to_currency_no_money($item_info->cost_price * 1.4))
							);?>

**Comment by hoangtu406** [comment]
I think you should go to ../application /libraries / Sale-lib.php.
Then, find method add_item (......).
About line 296; $price=$price!=null?$price:$item_info->unit_price;
Change to  $price=.... your calculation...

Hope this will help you.

**Comment by odiea** [comment]
Thanks for the tip. I had to go to controllers/items to save to get the updated price. Right now I am just trying to get that amount to show on the add Item form after I enter the cost price. Visual effects.

**Comment by daN4cat** [comment]
Was the idea to preload a default markup?

Maybe would be good to have the markup set it in store config and add something in the controller in case unit_price is 0, so the unit_price is created from the cost_price based on the default markup.
What do you think?

I remember another guy asked about a similar feature.

**Comment by odiea** [comment]
Yes. Currently I have it set as a field under General config. Yes, I set it up so that if the markup  is >0 it uses the Item markup and also read only placed on the Retail Price field In Add new and Update Item if markup is greater than 0.Of course it's not OSPOS standards. I am not worth a darn in programming or math so Instead of entering the 40% markup it was easier to use 1.40
![2016-05-14 04_36_53-christmas sweater store -- powered by os point of sale](https://cloud.githubusercontent.com/assets/10962177/15267958/8b4974be-198d-11e6-9c81-01f96931b537.png)

**Comment by daN4cat** [comment]
Looks good, maybe could be added as pre loaded default, but I would leave the retail price as something that can be retouched so people can do some manual rounding. 

BTW 1.40 is 40% markup, so you have done the correct maths there :+1: 

I would change the subject of this issue and reopen it.

**Comment by odiea** [comment]
I also considered not putting read only on the form but my issue is Employees not doing their jobs properly. This way someone with admin right can change that as necessary. It also changes the info in I think controller so changing the info in the form if markup is enabled really does nothing. I could not get the form to save any info so I placed it there.
$markup = $this->config->item('item_markup'); 
		if ($markup > 0)
			{
		$unit_price = $this->input->post('cost_price') * $markup;
			}
			else
			{
		$unit_price = $this->input->post('unit_price');
			}

**Comment by daN4cat** [comment]
OK, good point. Editing of that field should be permission based.

So if markup is set you would overwrite the unit_price anyway even if already set.
In the case I mentioned it would be better to do only if 0 so if somebody does a manual rounding it will be fine and retained. I know it's probably not your case.

Also I believe would be good to have the unit price automatically calculated during the editing and based on the input cost. That requires a bit of js.

I guess for my wife's case it would be handy because we had mistakes in the past, but markup varies according to supplier so for my case it would be better to fetch the markup from suppliers.

More to consider here.

**Comment by odiea** [comment]
Yes. That's why your the programmer and I am only the occasional tester these days.

**Comment by daN4cat** [comment]
I would say important contributor to the project :+1:
